backEnd/routes/auth.py

```diff
@@ -1,17 +1,187 @@
+            )
+
+        # Create JWT token if authentication is successful
+        access_token_expires = timedelta(minutes=ACCESS_TOKEN_EXPIRE_MINUTES)
+        access_token = create_access_token(
+            data={"sub": user["username"]}, expires_delta=access_token_expires
+        )
+
+        # Add successful login status code for frontend tracking (optional)
+        response.status_code = status.HTTP_200_OK
+
+        # Return the token with additional data
+        return {
+            "access_token": access_token,
+            "token_type": "bearer",
+            # Expiration in seconds for frontend handling
+            "expires_in": ACCESS_TOKEN_EXPIRE_MINUTES * 60,
+            "username": user["username"]  # just fro testing
+        }
+
+    except HTTPException as e:
+        # Log authentication errors (optional for tracking)
+        logging.error(f"Authentication failed for user {
+                      form_data.username}: {e.detail}")
+        raise e
+
+    except Exception as general_error:
+        # Log unexpected errors for server-side debugging
+        logging.error(f"Unexpected error during login: {general_error}")
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail="An unexpected error occurred. Please try again later."
+        )
+
+
+# signUp
+@router.post("/api/v1/auth/signUp", response_model=dict)
+async def sign_up(user: User):
+    try:
+        # Check if the user already exists in the database (by username or email)
+        existing_user = get_user_from_db(user.username)
+
+        if existing_user:
+            raise HTTPException(
+                status_code=status.HTTP_400_BAD_REQUEST,
+                detail="Username already registered"
+            )
+
+        # Optional: Check if email is already registered
+        existing_user_by_email = get_user_by_email(user.email)
+        if existing_user_by_email:
+            raise HTTPException(
+                status_code=status.HTTP_400_BAD_REQUEST,
+                detail="Email already registered"
+            )
+
+        # Hash the password before storing it
+        hashed_password = hash_password(user.password)
+
+        # Replace the plain password with the hashed one
+        user_with_hashed_password = User(
+            username=user.username,
+            email=user.email,
+            password=hashed_password
+        )
+
+        # Try to save the user with the hashed password to the database
+        try:
+            save_user_to_db(user_with_hashed_password)
+        except Exception as db_error:
+            logging.error(f"Error saving user to database: {db_error}")
+            raise HTTPException(
+                status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+                detail="An error occurred while saving the user to the database"
+            )
 
+        return {"message": "User successfully signed up"}
```

Comment:
Send me the created user with a token instead. So I can skip log in after sign up.
> always send the created/updated object in the response.